### PR TITLE
refactor(app): add next gen app feature flag

### LINKED ISSUE #8925: Create a Feature Flag to hide all of this Next Gen work

### PR DESCRIPTION
**Pull request by brenthagen**
# Overview

here's a next gen app feature flag and a `NextGenApp` component that will contain the new routes and navigation. for now, it only renders App Settings. `NextGenApp` has some temporary redirects that we'll need as long as the existing app is in place.

closes #8925

# Changelog

 - Added Next Gen App feature flag

# Review requests

toggle the feature flag back and forth

# Risk assessment

low

### CODE REVIEW

**Comment by smb2268** [comment]
I noticed that when you toggle the FF off, you get a blank screen and have to navigate to one of the options - should we route this back to the App Settings page automatically?
Screen Shot 2021-12-08 at 10 31 01 AM

**Comment by brenthagen** [comment]
> I noticed that when you toggle the FF off, you get a blank screen and have to navigate to one of the options - should we route this back to the App Settings page automatically? Screen Shot 2021-12-08 at 10 31 01 AM

yeah, we can add a redirect back to the app settings page in the "legacy" app

File: app/src/redux/config/constants.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ export const CONFIG_VERSION_LATEST: 1 = 1
 export const DEV_INTERNAL_FLAGS: DevInternalFlag[] = [
   'allPipetteConfig',
   'enableBundleUpload',
+  'nextGenApp',
```
Comment (b-cooper) [inline app/src/redux/config/constants.ts]: How about a blander name like "hierarchyReorganization" or something similar?

**Review by b-cooper — APPROVED** [review]
🎏

**Pull request merged**